crates/revm/src/db/states/bundle_state.rs

```diff
@@ -7,6 +7,185 @@ use revm_interpreter::primitives::{
     hash_map::{self, Entry},
     AccountInfo, Bytecode, HashMap, StorageSlot, B160, B256, KECCAK_EMPTY, U256,
 };
+use std::collections::{BTreeMap, BTreeSet, HashSet};
+
+/// This builder is used to help to facilitate the initialization of `BundleState` struct
+#[derive(Debug, Default)]
+pub struct BundleBuilder {
+    states: HashSet<B160>,
+    state_original: HashMap<B160, AccountInfo>,
+    state_present: HashMap<B160, AccountInfo>,
+    state_storage: HashMap<B160, HashMap<U256, (U256, U256)>>,
+
+    reverts: BTreeSet<(usize, B160)>,
+    revert_capacity: usize,
+    revert_account: HashMap<(usize, B160), Option<Option<AccountInfo>>>,
+    revert_storage: HashMap<(usize, B160), Vec<(U256, U256)>>,
+
+    contracts: HashMap<B256, Bytecode>,
+}
+
+impl BundleBuilder {
+    /// Create builder instance
+    ///
+    /// `revert_capacity` indicates the size of BundleState `reverts` field
+    pub fn new(revert_capacity: usize) -> Self {
```

Comment:
Would do it a little bit differently and use `RangeInclusive<u64>` here. Where u64 would represent the block number.

The idea is to have API with block number as this is the number that is expected, `block_index` seems unnatural.

And you have can iterate over it with when creating reverts.

Other things seem good